### PR TITLE
enhancement(prometheus): support prometheus2 .yml rule file format

### PR DESCRIPTION
**Pull request by dpavle**

### CODE REVIEW

File: roles/prometheus/templates/prometheus.yml.j2 [hunk 1]
```diff
@@ -20,6 +20,7 @@ remote_read:
 {% if not prometheus_agent_mode and prometheus_alert_rules_files != [] %}
 rule_files:
   - {{ prometheus_config_dir }}/rules/*.rules
+  - {{ prometheus_config_dir }}/rules/*.rules.yml
```
[the remark this answers is not in the document]
Comment (dpavle) [reply roles/prometheus/templates/prometheus.yml.j2:23]: Yep, that makes sense.

**Comment by SuperQ** [comment]
Please also update the updated docs.

**Review by SuperQ — APPROVED** [review]

**Review by gardar — CHANGES_REQUESTED** [review]
If we want to support both `.rules` and `.yml` as proposed by this PR then both should be listed in the docs and argument specs.

But if we want to drop `.rules` then that should be done clearly with a warning in the preflight.

Also `.yaml` should be supported since both`.yml` and `.yaml` are valid yaml file extensions.

**Comment by dpavle** [comment]
> If we want to support both `.rules` and `.yml` as proposed by this PR then both should be listed in the docs and argument specs.
> 
> But if we want to drop `.rules` then that should be done clearly with a warning in the preflight.
> 
> Also `.yaml` should be supported since both`.yml` and `.yaml` are valid yaml file extensions.

I'm for removing the old `.rules` format. It's essentially been deprecated [since 2017](https://prometheus.io/blog/2017/06/21/prometheus-20-alpha3-new-rule-format/) and it doesn't work in any recent version of Prometheus as far as I'm aware. 

As for the warning in the preflight, sure, that could be done. I'm just not sure how exactly, do we fail if any `.rules` files are found in `/rules`?  

And yes, `.yaml` should also be supported.

**Comment by SuperQ** [comment]
We probably want to keep the `*.rules` handling, at least from a configuration loading perspective, around in order to maintain backwards compatibility. Otherwise existing deployments that are pushing yaml to files named `.rules` will break.

**Comment by gardar** [comment]
> We probably want to keep the `*.rules` handling, at least from a configuration loading perspective, around in order to maintain backwards compatibility. Otherwise existing deployments that are pushing yaml to files named `.rules` will break.

That's why I suggested a error in the preflight, but we could also make that a warning for now and then drop `*.rules` support in the future.
Or we could handle it automatically by adding a task that renames `.rules` to `.yml`

**Comment by dpavle** [comment]
> > We probably want to keep the `*.rules` handling, at least from a configuration loading perspective, around in order to maintain backwards compatibility. Otherwise existing deployments that are pushing yaml to files named `.rules` will break.
> 
> That's why I suggested a error in the preflight, but we could also make that a warning for now and then drop `*.rules` support in the future. Or we could handle it automatically by adding a task that renames `.rules` to `.yml`

A warning works. If there are any YAML formatted files with the .rules extension there will be a warning message in the preflight and the rule validation will pass normally. On the other hand, if there are any rules in the old .rules format, the warning message will show in the preflight and promtool check validation will catch them and fail in a later step.

**Review by gardar — APPROVED** [review]
LGTM

**Comment by gardar** [comment]
@dpavle any chance you could refactor this so that we can get it merged?

**Comment by dpavle** [comment]
> @dpavle any chance you could refactor this so that we can get it merged?

sure, I'll take a look at it

**Comment by gardar** [comment]
@SuperQ any outstanding issues or are we good to go?

**Review by gardar — APPROVED** [review]

**Review by SuperQ — APPROVED** [review]
Yup, LGTM

**Pull request merged**